pkg/gui/branches_panel.go

```diff
@@ -9,154 +9,346 @@ import (
-			return gui.createErrorPanel(g, err.Error())
-		}
-		return gui.refreshSidePanels(g)
-	})
+
+	err := gui.createPromptPanel(v, gui.Tr.SLocalize("BranchName")+":",
+		func(g *gocui.Gui, v *gocui.View) error {
+
+			err := gui.GitCommand.Checkout(gui.trimmedContent(v), false)
+			if err != nil {
+				err = gui.createErrorPanel(err.Error())
+				if err != nil {
+					gui.Log.Errorf("Failed to create error panel at handleCheckoutByName: %s\n", err)
+					return err
+				}
+
+				return nil
+			}
+
+			gui.refresh()
+
+			return nil
+		})
+	if err != nil {
+		gui.Log.Errorf("Failed to create prompt panel at handleCheckoutByName: %s\n", err)
+		return err
+	}
+
 	return nil
 }
 
+// handleNewBranch is called when a user wants to create a new branch.
+// g and v are passed by the gocui library but only v is used.
+// If something goes wrong it returns an error.
 func (gui *Gui) handleNewBranch(g *gocui.Gui, v *gocui.View) error {
+
 	branch := gui.State.Branches[0]
 	message := gui.Tr.TemplateLocalize(
 		"NewBranchNameBranchOff",
 		Teml{
 			"branchName": branch.Name,
 		},
 	)
-	gui.createPromptPanel(g, v, message, func(g *gocui.Gui, v *gocui.View) error {
-		if err := gui.GitCommand.NewBranch(gui.trimmedContent(v)); err != nil {
-			return gui.createErrorPanel(g, err.Error())
-		}
-		gui.refreshSidePanels(g)
-		return gui.handleBranchSelect(g, v)
-	})
+
+	err := gui.createPromptPanel(v, message,
+		func(g *gocui.Gui, v *gocui.View) error {
+
+			err := gui.GitCommand.NewBranch(gui.trimmedContent(v))
+			if err != nil {
+
+				err = gui.createErrorPanel(err.Error())
+				if err != nil {
+					gui.Log.Errorf("Failed to create error panel at handleNewBranch: %s\n", err)
+					return err
+				}
+
+				return nil
+			}
+
+			gui.refresh()
+
+			err = gui.handleBranchSelect(v)
+			if err != nil {
+				gui.Log.Errorf("Failed to handleBranchSelect at handleNewBranch: %s\n", err)
+				return err
+			}
+
+			return nil
+		})
+	if err != nil {
+		gui.Log.Errorf("Failed to create prompt panel at handleNewBranch: %s\n", err)
+		return err
+	}
+
 	return nil
 }
 
+// handleDeleteBranch gets called when the user wants to normally delete a
+// branch.
 func (gui *Gui) handleDeleteBranch(g *gocui.Gui, v *gocui.View) error {
-	return gui.deleteBranch(g, v, false)
+	return gui.deleteBranch(v, false)
 }
 
+// handleForceDeleteBranch gets called when the user wants to force delete a
+// branch
 func (gui *Gui) handleForceDeleteBranch(g *gocui.Gui, v *gocui.View) error {
-	return gui.deleteBranch(g, v, true)
+	return gui.deleteBranch(v, true)
 }
 
-func (gui *Gui) deleteBranch(g *gocui.Gui, v *gocui.View, force bool) error {
+// deleteBranch gets called when the user wants to delete a branch.
+// v is passed for ease sake, force is to indicate if it must be force
+// deleted.
+// If anything goes wrong, it returns an error.
+func (gui *Gui) deleteBranch(v *gocui.View, force bool) error {
+
+	var messageId string
```

Comment:
I don't think messageId should be declared up here, because there is no context until later that shows what its purpose is. By declaring the variable right before its value is assigned, it's easier to see what the function is doing.

Reply:
Again, C styling got me used to this